### ISSUE TITLE
Provide toggle to display transmission lines in the map

### ISSUE THREAD

**Issue by trvrb** [nextstrain/auspice#1095]
**Context**  
With SARS-CoV-2 jumping around the globe so quickly, showing transmission lines on the map can be overwhelming in terms of information content and detract from overall understanding. See for example:

<img width="1134" alt="lines" src="https://user-images.githubusercontent.com/1176109/80832435-807ea380-8ba1-11ea-9418-b0972f8c21f3.png">

There are frequently situations where we want "traits" reconstruction on phylogeny, but don't want to show these lines on the map.

**Description**   

Include a toggle in the sidebar under "Map Options" directly below "Geographic Resolution" that says "Show transmission lines". When toggled on, we'd have the current behavior. When toggled off, the deme circles / pie charts would remain but no transmission lines would be drawn on the map.

In addition, this toggle should update the URL and (eventually) this "show transmission lines" default behavior should be included in the Auspice JSON as something like:

```
    "display_defaults": {
      "transmission_lines": false
    }, 
```

But would be good to get working as UI element and URL before bothering to include in JSON.

**Comment by marcaaron** [comment]
Would like to submit a PR for this one. Here's my proposed solution:

1. Add new type `TOGGLE_TRANSMISSION_LINES` and control state field `showTransmissionLines`
1. Create a new control component `TransmissionLines`
1. Modify `maybeRemoveAllDemesAndTransmissions` so that a change in the transmission lines setting prop will trigger a redraw
1. Add a new parameter to `drawDemesAndTransmissions` called `showTransmissionLines`
1. Then inside `drawDemesAndTransmissions` conditionally draw the lines or not by applying a stroke of `1` or `0`.

<details>
<summary>Code Example</summary>

```diff
diff --git a/src/components/map/mapHelpers.js b/src/components/map/mapHelpers.js
index d3f098c0..e534045a 100644
--- a/src/components/map/mapHelpers.js
+++ b/src/components/map/mapHelpers.js
@@ -137,7 +137,8 @@ export const drawDemesAndTransmissions = (
   numDateMax,
   pieChart, /* bool */
   geoResolution,
-  dispatch
+  dispatch,
+  showTransmissionLines
 ) => {

   // add transmission lines
@@ -164,7 +165,7 @@ export const drawDemesAndTransmissions = (
     .attr("stroke-opacity", 0.6)
     .attr("stroke-linecap", "round")
     .attr("stroke", (d) => { return d.color; })
-    .attr("stroke-width", 1);
+    .attr("stroke-width", showTransmissionLines ? 1 : 0);

   const demeMultiplier = demeCountMultiplier / Math.sqrt(_max([nodes.length, demeCountMinimum]));
   let demes;
```
</details>

**Here are some screenshots showing what the change would look like:**
![Screen Shot 458](https://user-images.githubusercontent.com/32969087/80871731-cea7ab80-8c62-11ea-89b6-f689556708cf.png)
![Screen Shot 457](https://user-images.githubusercontent.com/32969087/80871733-cfd8d880-8c62-11ea-8ae5-80394ab94fe0.png)

**Comment by jameshadfield** [comment]
Thanks @marcaaron, that would be great -- the solution in the screenshots looks pretty good.

Drawing transmissions can be computationally expensive, so long-term I'd favor not calculating them & not drawing them. This is _a lot more work_ than the `stroke-width=0` solution so separating this into another issue/PR would make for more atomic PRs.

How we store on/off state (e.g. `showTransmissionLines`) should be keyed off of geographic-resolution, and so should the JSON-provided `display_defaults`, so that different resolutions (e.g. country vs region) can have different toggle states. (I think the `display_defaults`, and the URL query, can be different issues/PRs.)

**Comment by marcaaron** [comment]
Good stuff. Yep, I agree this is definitely not the most ideal way, but should get us there quickly :) 
I opened a draft PR so we can discuss improvements there.

**Comment by trvrb** [comment]
Bumping this issue. 

First off, thank you so much @marcaaron for your work on this. Hugely helpful. Addition context in PR discussion here: https://github.com/nextstrain/auspice/pull/1103.

With merging of https://github.com/nextstrain/auspice/pull/1147 we have a toggle on the sidebar to show/hide transmission lines:

![lines](https://user-images.githubusercontent.com/1176109/83580853-f902b800-a4f1-11ea-9002-d70b95f4e94b.png)

This is already quite useful, but we'd like the ability to both control display via URL and also via auspice config. When loading we want to look for:

```
 "display_defaults": {
      "transmission_lines": false
    }, 
```

to set this toggle on/off.

Currently, we'd like transmission lines to be off for the default view at https://nextstrain.org/ncov/global as it's overwhelming having so many lines shown on the map. But after zooming in a bit it becomes more relevant to show transmission lines.

**Comment by jameshadfield** [comment]
Agreed. I'm going to allow the default setting to be taken from JSON defaults now, and will make separate issues for other to-do items such as URL queries, improved performance etc.

**Comment by jameshadfield** [comment]
Closing this issue as the toggle has been implemented (#1147) as well as the ability to set this via the JSON's `display_defaults` (#1152) and performance improvements (#1153). Separate issues have been / will be created for remaining to-do items (e.g. #1154)
